src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -52,25 +69,36 @@ export const Configurator = ({
       defaultMessage: "I don't want to use the time filter",
     }
   );
-  const [language, setLanguage] = useState<string>(() => {
-    const currentLanguage = queryString.getQuery().language;
-    if (languages.includes(currentLanguage)) {
-      return currentLanguage;
-    }
-    return languages[0];
-  });
+  const indexedViewsService = type?.indexedViewsService;
+  const [selectedIndexedView, setSelectedIndexedView] = useState<string>();
+  const [indexedViews, setIndexedViews] = useState<DatasetIndexedView[]>([]);
+  const [isLoadingIndexedViews, setIsLoadingIndexedViews] = useState(false);
+
+  useEffect(() => {
+    const getIndexedViews = async () => {
+      if (indexedViewsService) {
+        setIsLoadingIndexedViews(true);
+        const fetchedIndexedViews = await indexedViewsService.getIndexedViews(baseDataset);
+        setIsLoadingIndexedViews(false);
+        setIndexedViews(fetchedIndexedViews || []);
+      }
+    };
+
+    getIndexedViews();
+  }, [indexedViewsService, baseDataset]);
 
   const submitDisabled = useMemo(() => {
     return (
-      timeFieldName === undefined &&
-      !(
-        languageService.getLanguage(language)?.hideDatePicker ||
-        dataset.type === DEFAULT_DATA.SET_TYPES.INDEX_PATTERN
-      ) &&
-      timeFields &&
-      timeFields.length > 0
+      isLoadingIndexedViews ||
+      (timeFieldName === undefined &&
+        !(
+          languageService?.getLanguage(language)?.hideDatePicker ||
+          dataset.type === DEFAULT_DATA.SET_TYPES.INDEX_PATTERN
+        ) &&
+        timeFields &&
+        timeFields.length > 0)
     );
-  }, [dataset, language, timeFieldName, timeFields, languageService]);
+  }, [dataset, language, timeFieldName, timeFields, languageService, isLoadingIndexedViews]);
```

Comment:
nit: do we need to prevent the user from clicking submit if the indexed view is still being loaded? they should be able to submit and would be able to query for results with or without indexed views so it seems like preventing the user from submitting would make it seem like an indexed view is required to query.

Reply:
Removing the check